File: ItCameDroid-AS/app/src/main/java/br/odb/giovanni/engine/Tile.java
package br.odb.giovanni.engine;

import android.content.res.Resources;
import android.graphics.Bitmap;

public class Tile extends br.odb.giovanni.engine.Bitmap {
    private int type;

    public Tile(Resources resources, int resId) {
        super(resources, resId);

        type = resId;
    }

    public Tile(int baseTypeId, Bitmap androidBitmap) {
        super(androidBitmap);

        type = baseTypeId;
    }

    public int getType() {
        return type;
    }

    public void setTile(int type, android.graphics.Bitmap tile) {
        super.setAndroidBitmap(tile);
        this.type = type;
    }
}

File: ItCameDroid-AS/app/src/main/java/br/odb/giovanni/engine/Tile.kt
package br.odb.giovanni.engine

import android.content.res.Resources

class Tile : Bitmap {
    var type: Int
        private set

    constructor(resources: Resources?, resId: Int) : super(resources, resId) {
        type = resId
    }

    constructor(baseTypeId: Int, androidBitmap: android.graphics.Bitmap?) : super(androidBitmap) {
        type = baseTypeId
    }

    fun setTile(type: Int, tile: android.graphics.Bitmap?) {
        super.androidBitmap = tile
        this.type = type
    }
}